lib/va_profile/models/v3/validation_address.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require_relative 'base_address'
+require 'common/hash_helpers'
+
+module VAProfile
+  module Models
+    module V3
+      # Model for addresses sent and received from the VA profile address validation API
+      # AddressValidationV is used for ProfileServiceV3 and ContactInformationV2
+      class ValidationAddress < V3::BaseAddress
+        # Convert a ValidationAddress into a hash that can be sent to the address validation
+        # API
+        # @return [Hash] hash that is formatted for POSTing to address validation API
+        def address_validation_req
+          Common::HashHelpers.deep_remove_blanks(
+            address: attributes.slice(
+              :address_line1,
+              :address_line2,
+              :address_line3
+            ).deep_transform_keys { |k| k.to_s.camelize(:lower) }.merge(
+              intPostalCode: @international_postal_code,
+              cityName: @city,
+              zipCode5: @zip_code,
+              zipCode4: @zip_code_suffix,
+              country: { countryCodeISO2: @country_code_iso2, countryCodeISO3: @country_code_iso3,
+                         countryName: @country_name, countryCodeFIPS: @country_code_fips },
+              state: { stateCode: @state_code, stateName: @state_name },
+              province: { provinceName: @province_name, provinceCode: @province_code },
+              addressPOU: @address_pou
+            )
+          )
+        end
+
+        # @return [VAProfile::Models::V3::ValidationAddress] validation address model created from
+        #   address validation API response
+        def self.build_from_address_suggestion(address_suggestion_hash)
+          address_type = address_suggestion_hash['address_type'].upcase
+          attributes = {
+            address_line1: address_suggestion_hash['address_line1'],
+            address_line2: address_suggestion_hash['address_line2'],
+            address_line3: address_suggestion_hash['address_line3'],
+            address_type:,
+            city: address_suggestion_hash['city_name'],
+            country_name: address_suggestion_hash['country']['country_name'],
+            country_code_iso3: address_suggestion_hash['country']['country_code_iso3']
+          }.merge(regional_attributes(address_type, address_suggestion_hash))
+
+          new(attributes)
+        end
+
+        def self.regional_attributes(address_type, address_hash)
```

Comment:
In this method, you probably want to address issues where `undefined method [] for nil` could be raised if parts of the hash don’t exist (ex: address_hash['province'] or address_hash['state'] is nil). `.dig` is probably a safer option by being more safe about retrieving nested values, returning nil if the key doesn’t exist.

Reply:
Province or state will always be null. Province is used for outside the US.

Reply:
But what if the key doesn't exist? Should we account for that using `.dig` which is a safer option here?

Reply:
I'll put a dig just incase.

Reply:
@LindseySaari DONE!